source/utils/timed-out.ts

```diff
@@ -3,6 +3,9 @@ import {ClientRequest, IncomingMessage} from 'http';
 import {Delays} from './types';
 import unhandler from './unhandle';
 
+// Help TypeScript to understand how we are passing arguments thru setImmediate
+declare function setImmediate<T extends any[]>(callback: (arg0: number, ...args: T) => void, arg0: number, ...args: T): NodeJS.Immediate;
```

Comment:
Is this something you could upstream to the Node.js types?

Reply:
Unfortunately it requires a very recent version of TypeScript, and they generally don't accept PRs with that since it breaks for a lot of users using DefinitivelyTyped 😞

Reply:
Alright, I can you add a TODO comment about upstreaming it at some point?

Reply:
We'll also be running into this when it's upstreamed 😬 

https://github.com/microsoft/TypeScript/issues/26113

Do you want to me to **a)**

Declare `setImmediate` to something that could be upstreamed:

```ts
// TODO: Upstream this once DefinitelyTyped accepts `T extends any[]`
declare function setImmediate<T extends any[]>(callback: (...args: T) => void, ...args: T): NodeJS.Immediate;
```

and then ignore the error where we use it:

```ts
// @ts-ignore https://github.com/microsoft/TypeScript/issues/26113
immediate = setImmediate(callback, delay, ...args);
```

or **b)**

Keep our specialised functions and add both comments there:

```ts
// TODO: Remove `arg0` when https://github.com/microsoft/TypeScript/issues/26113 is fixed
// TODO: Upstream this once DefinitelyTyped accepts `T extends any[]`
declare function setImmediate<T extends any[]>(callback: (arg0: number, ...args: T) => void, arg0: number, ...args: T): NodeJS.Immediate;
```

Reply:
I am not really a fan of manually overriding the function, while it it how *we* use it. Thoughts on just `// @ts-ignore` with a TODO to remove once the issue is closed?

Reply:
Let's go with just a tsignore comment.